### PR TITLE
Fetch data and update Redux store

### PR DESCRIPTION
**Pull request by mryeminthu**
Upon first render fetch data from the SpaceX API endpoints:
Rockets: https://api.spacexdata.com/v4/rockets

### CODE REVIEW

File: src/components/Rockets.jsx [hunk 1]
```diff
@@ -1,38 +1,65 @@
...
-      </div>
-    ))}
-  </div>
-);
+  useEffect(() => {
+    fetch('https://api.spacexdata.com/v4/rockets')
+      .then((response) => response.json())
+      .then((data) => setRocketData(data))
+      .catch(() => {});
+  }, []);
```
Comment (lily4178993) [inline src/components/Rockets.jsx]: To meet the requirement, you should modify the **`Rockets`** component to dispatch an action to store the fetched data in the Redux store. Here is an example:

```js
// ...rest of your code...
useEffect(() => {
    fetch('https://api.spacexdata.com/v3/rockets')
      .then((response) => response.json())
      .then((data) => {
        // Dispatch an action to store the fetched data in Redux
        dispatch(fetchRockets(data));
        setRocketData(data);
      })
      .catch(() => {});
  }, [dispatch]);
```
// ...rest of your code...

File: src/redux/rocketActions.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export const reserveRocket = (rocketId) => ({
+  type: 'RESERVE_ROCKET',
+  payload: rocketId,
+});
+
+export const cancelReservation = (rocketId) => ({
+  type: 'CANCEL_RESERVATION',
+  payload: rocketId,
+});
```
Comment (lily4178993) [inline src/redux/rocketActions.js:14]: Following the same logic, you have to create an action creator **`fetchRockets`** in your **rocketActions.js** file to handle storing the fetched data in the Redux store. Here is an example:

```js
// Action type to store fetched rockets
export const FETCH_ROCKETS = 'FETCH_ROCKETS';

// Action creator to store fetched rockets in Redux
export const fetchRockets = (rockets) => ({
  type: FETCH_ROCKETS,
  payload: rockets,
});

// ... Rest of your code ...
```

File: src/redux/rocketReducer.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    },
+    {
+      id: 2,
+      name: 'Falcon 2',
+      description: 'Description of Falcon 2',
+      reserved: false,
+    },
+  ],
+};
+
```
Comment (lily4178993) [inline src/redux/rocketReducer.js]: Update your **rocketReducer.js** to handle the **FETCH_ROCKETS** action and store the rocket data in the Redux store. Here is an example: 
```js
// Import the action type
import { FETCH_ROCKETS } from './rocketActions';

const initialState = {
  rockets: [],
};

const rocketReducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_ROCKETS:
      return { ...state, rockets: action.payload };

    // ... Other reducer cases ...
  }
};

export default rocketReducer;

```

**Review by lily4178993 — CHANGES_REQUESTED** [review]
Hi @mryeminthu

I've reviewed the code you shared for the Rockets components. Overall, it's looking good, but there are a couple of points to consider for improvement:

1. **Storing Fetched Data:**
It appears that the code does not store the fetched data from the Rockets endpoint in the Redux store. To meet the requirement, you should dispatch an action to store the selected data in Redux, including fields such as id, name, type, and flickr_images.

2. **Redux Actions and Reducer:**
Ensure that you have the necessary Redux action creators and reducers to handle the data fetching and storage process correctly. The **`rocketActions.js`** and **`rocketReducer.js`** files need to be updated.

Once these adjustments are made, the code should align with the specified requirements for data fetching and storage in the Redux store.

Feel free to reach out if you have any questions or need further assistance with the changes. Keep up the good work!

**Review by lily4178993 — CHANGES_REQUESTED** [review]
Hi @mryeminthu, please make sure to correctly configure the store by importing the **rocketsReducer** from the **rocketsSlice.js** file in the **store**. 

The yellow boxes is what you have to add in order to configure your store:
Capture d’écran (390)

- Before the configuration of the store:

Capture d’écran (391)

- Screenshot after correclty configured the store

Capture d’écran (389)


Thanks!

**Comment by mryeminthu** [comment]
Hello please just check my last pull request. What you mention is included in it.

**Pull request closed without merging**